hypergan/losses/base_loss.py

```diff
@@ -45,6 +53,8 @@ def create(self):
         self.metrics = self.metrics or sample_metrics
 
         self.sample = [d_loss, g_loss]
+        self.d_loss = d_loss
```

Comment:
These are probably better variable names for Loss instance methods.  I think the .sample pattern starts to fall apart for loss terms(we already exclude access to self.ops).